kernel/kernel-api/src/test/scala/io/delta/kernel/internal/files/ParsedLogDataSuite.scala

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal.files
+
+import java.util.Optional
+
+import io.delta.kernel.data.{ColumnarBatch, ColumnVector}
+import io.delta.kernel.internal.files.ParsedLogData.{ParsedLogCategory, ParsedLogType}
+import io.delta.kernel.internal.util.FileNames
+import io.delta.kernel.test.MockFileSystemClientUtils
+import io.delta.kernel.types.StructType
+import io.delta.kernel.utils.FileStatus
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class ParsedLogDataSuite extends AnyFunSuite with MockFileSystemClientUtils {
+
+  private val emptyColumnarBatch = new ColumnarBatch {
+    override def getSchema: StructType = null
+    override def getColumnVector(ordinal: Int): ColumnVector = null
+    override def getSize: Int = 0
+  }
+
+  /////////////
+  // General //
+  /////////////
+
+  test("Throws on unknown log file") {
+    val fileStatus = FileStatus.of("unknown", 0, 0)
+    val exMsg = intercept[IllegalArgumentException] {
+      ParsedLogData.forFileStatus(fileStatus)
+    }.getMessage
+    assert(exMsg.contains("Unknown log file type"))
+  }
+
+  test("Throws on version < 0") {
+    val exMsg = intercept[IllegalArgumentException] {
+      ParsedLogData.forInlineData(-1, ParsedLogType.RATIFIED_INLINE_COMMIT, emptyColumnarBatch)
+    }.getMessage
+    assert(exMsg.contains("version must be non-negative"))
+  }
+
+  test("Throws on both fileStatusOpt and inlineDataOpt present") {
+    val fileStatusOpt = Optional.of(deltaFileStatus(5))
+    val inlineDataOpt = Optional.of(emptyColumnarBatch)
+    val exMsg = intercept[IllegalArgumentException] {
+      new ParsedLogData(10, ParsedLogType.PUBLISHED_DELTA, fileStatusOpt, inlineDataOpt)
+    }.getMessage
+    assert(exMsg.contains("Exactly one of fileStatusOpt or inlineDataOpt must be present"))
+  }
+
+  test("Throws on both fileStatusOpt and inlineDataOpt empty") {
+    val exMsg = intercept[IllegalArgumentException] {
+      new ParsedLogData(10, ParsedLogType.PUBLISHED_DELTA, Optional.empty(), Optional.empty())
+    }.getMessage
+    assert(exMsg.contains("Exactly one of fileStatusOpt or inlineDataOpt must be present"))
+  }
+
+  ////////////
+  // Deltas //
```

Comment:
should there be a test for `RATIFIED_INLINE_COMMIT`?

Reply:
Added!